### PR TITLE
use origin for the notification message and for site settings and cho…

### LINKED ISSUE #3877: Regression : session.setPermissionRequestHandler callback is broken
**Did you search for similar issues before submitting this one?**
Yes

**Describe the issue you encountered:**
When testing http://www.bennish.net/web-notifications.html during a test run, it was noticed that the notifications were not displaying. 

**Expected behavior:**
The notifications should work.

- Platform (Win7, 8, 10? macOS? Linux distro?):
Windows 7 ia-32 and OS X

- Brave Version:
0.12.1

- Steps to reproduce:
    1. Visit http://www.bennish.net/web-notifications.html
    2. Trigger a notification

- Screenshot if needed:

- Any related issues:

### PR DESCRIPTION
**Pull request by bridiver**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Test Plan:
1. Go to https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/Using_geolocation
2. notification should display for https://mdn.mozillademos.org vs page origin of http://developer.mozilla.org
3. change tabs or open a new tab
4. notification should not appear
5. return to the original tab
6. notification should be displayed

The changes for Brave Browser notifications aren't currently  testable because we don't get the mainFrameUrl for notifications yet, but I have tested them with a local hack

don't allow Brave to display notifications without permission
remove unnecessary password copied notifications
cc @bbondy @diracdeltas
fix #3877

### CODE REVIEW

**Comment by bridiver** [comment]
fix #3962

**Comment by diracdeltas** [comment]
this causes some of the notificationBar tests to fail

File: app/filtering.js [hunk 1]
```diff
@@ -367,7 +374,7 @@ function registerPermissionHandler (session, partition) {
       cb(result)
       if (persist) {
         // remember site setting for this host over http(s)
-        appActions.changeSiteSetting(getOrigin(url), permission + 'Permission', result, isPrivate)
+        appActions.changeSiteSetting(origin, permission + 'Permission', result, isPrivate)
```
Comment (diracdeltas) [inline app/filtering.js]: need to strip the trailing `/` from origin for backwards compatibility; i think this is why tests are breaking

Comment (diracdeltas) [inline app/filtering.js]: hence why `getOrigin` was necessary before

**Pull request merged**